stories/charts.js

```diff
@@ -102,34 +103,34 @@ const customizedColorMap = [
 
 storiesOf('Chart', module)
   .add('SummaryHorizontalBarChart', () => (
-    <SummaryHorizontalBarChart data={virusData} title='bar chart title' showPercentage={false} />
+    <SummaryHorizontalBarChart data={virusData} title='bar chart title' showPercentage={false} totalCount={virusDataTotal}/>
   ))
   .add('SummaryHorizontalBarChart with single color and percentage', () => (
-    <SummaryHorizontalBarChart data={virusData} title='bar chart title' color='#3283c8' />
+    <SummaryHorizontalBarChart data={virusData} title='bar chart title' color='#3283c8'  totalCount={virusDataTotal} />
   ))
   .add('SummaryHorizontalBarChart with customized colors', () => (
-    <SummaryHorizontalBarChart data={virusData} title='bar chart title' showPercentage={false} useCustomizedColorMap customizedColorMap={customizedColorMap} />
+    <SummaryHorizontalBarChart data={virusData} title='bar chart title' showPercentage={false} useCustomizedColorMap customizedColorMap={customizedColorMap}  totalCount={virusDataTotal}/>
   ))
   .add('SummaryPieChart', () => (
-    <SummaryPieChart data={virusData} title='pie chart title' showPercentage />
+    <SummaryPieChart data={virusData} title='pie chart title' showPercentage  totalCount={virusDataTotal}/>
   ))
   .add('SummaryPieChart with customized colors', () => (
-    <SummaryPieChart data={virusData} title='pie chart title' showPercentage useCustomizedColorMap customizedColorMap={customizedColorMap} />
+    <SummaryPieChart data={virusData} title='pie chart title' showPercentage useCustomizedColorMap customizedColorMap={customizedColorMap} totalCount={virusDataTotal} />
   ))
   .add('SummaryFullPieChart', () => (
-    <SummaryPieChart data={virusData} title='pie chart title' innerRadius={0} showPercentage />
+    <SummaryPieChart data={virusData} title='pie chart title' innerRadius={0} showPercentage totalCount={virusDataTotal} />
   ))
   .add('SummaryChartGroup', () => (
-    <SummaryChartGroup summaries={summaries} width={1010} />
+    <SummaryChartGroup summaries={summaries} width={1010} totalCount={10000} />
   ))
   .add('SummaryChartGroup with only showing 2 at first', () => (
-    <SummaryChartGroup summaries={summaries} width={1010} maximumDisplayItem={2} />
+    <SummaryChartGroup summaries={summaries} width={1010} maximumDisplayItem={2} totalCount={1000} />
   ))
   .add('SummaryChartGroup with a Locked Chart', () => (
-    <SummaryChartGroup summaries={lockedSummaries} width={1010} lockValue={-1} lockMessage='This chart is locked!' />
+    <SummaryChartGroup summaries={lockedSummaries} width={1010} lockValue={-1} lockMessage='This chart is locked!' totalCount={1000} />
```

Comment:
This logic needs be able to handle lock values `-1` by default, we cannot hard code this total count if the data returned has lock values

Reply:
normally in the data portal this value will be calculated automatically based on props.totalCount but i will remove all of the hard coded values and implement the old logic if the totalCount is omitted as you suggested

Reply:
Oh yeah I get it, thanks. 

But I'd suggest let's revisit the design in this ticket with its owner, before you spending some effort in here. I just don't want them to be wasted